### PR TITLE
Internal java solution for CSES ferris wheel

### PR DESCRIPTION
**Pull request by nathangong**

### CODE REVIEW

File: solutions/cses-1090.mdx [hunk 1]
```diff
@@ -81,4 +82,53 @@ int main(){
 
 </CPPSection>
 
+<JavaSection>
+
+<Warning>
+Java will TLE on test case #10.
```
Comment (brad182) [inline solutions/cses-1090.mdx:88]: dang java so slow its fine for usaco but codeforces and other websites nah cuz no extra time :(

File: solutions/cses-1090.mdx [hunk 2]
```diff
@@ -81,4 +82,53 @@ int main(){
...
+        int x = io.nextInt();
+
+        int[] weights = new int[n];
+        for (int i = 0; i < n; i++) {
+            weights[i] = io.nextInt();
+        }
+        Arrays.sort(weights);
+
+        int ans = 0;
+        int i = 0; // Left pointer
```
Comment (brad182) [inline solutions/cses-1090.mdx:109]:
```suggestion
        int i = 0;  // left pointer
```

File: solutions/cses-1090.mdx [hunk 3]
```diff
@@ -81,4 +82,53 @@ int main(){
...
+        int j = n - 1; // Right pointer
```
Comment (brad182) [inline solutions/cses-1090.mdx:110]:
```suggestion
        int j = n - 1;  // right pointer
```

File: solutions/cses-1090.mdx [hunk 4]
```diff
@@ -81,4 +82,53 @@ int main(){
...
+        while (i <= j) {
+            ans++; // Increment number of gondolas used
```
Comment (brad182) [inline solutions/cses-1090.mdx:112]:
```suggestion
            ans++;  // Increment number of gondolas used
```

**Review by brad182 — CHANGES_REQUESTED** [review]
just need double spaces for the comments, the rest is good

File: solutions/cses-1090.mdx (hunk 1 above)
Comment (dongliu0426) [reply solutions/cses-1090.mdx:88]: try using fast io?

Comment (brad182) [reply solutions/cses-1090.mdx:88]: isn't kattio already fast io? it uses buffered reader which is the fastest way I think?

Comment (brad182) [reply solutions/cses-1090.mdx:88]: or is this something that i don't know about

Comment (nwatx) [reply solutions/cses-1090.mdx:88]: he is referring to https://usaco.guide/general/fast-io/#method-5---inputstream i believe

Comment (nathangong) [reply solutions/cses-1090.mdx:88]: rip FastIO still TLE's on tc 10

Comment (dongliu0426) [reply solutions/cses-1090.mdx:88]: hmmm I see java solutions that pass with scanner lol https://cses.fi/problemset/hack/1090/entry/1990829/

Comment (nathangong) [reply solutions/cses-1090.mdx:88]: Can you send a pic? For some reason the link just shows a blank screen

Comment (dongliu0426) [reply solutions/cses-1090.mdx:88]: i'll send the code:

<details>

<summary>
Code
</summary>

```java
import java.util.Arrays;
import java.util.Scanner;
 
public class CSES1090 {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		
		while (input.hasNext()) {
			int n = input.nextInt();
			int x = input.nextInt();
			input.nextLine();
			String[] pStr = input.nextLine().split(" ");
			Integer[] p = new Integer[n];
			
			for (int i = 0; i < p.length; ++i) {
				p[i] = Integer.decode(pStr[i]);
			}
			
			Arrays.sort(p);
			
			int ans = 0;
			for (int i = 0, j = p.length - 1; i <= j; --j) {
				if (i != j && x - p[j] >= p[i]) {
					++i;
				}
				++ans;
			}
			
			System.out.println(ans);
		}
	}
}
```

</details>

Comment (brad182) [reply solutions/cses-1090.mdx:88]: this is just bad the solutions to problems should be on the algorithm and not on how you use IO

Comment (nathangong) [reply solutions/cses-1090.mdx:88]: This uses the same logic as my code, I guess the way they read in the array is faster?

Comment (nathangong) [reply solutions/cses-1090.mdx]: btw I fixed the TLE issue so it now passes the all test cases

**Comment by SansPapyrus683** [comment]
just asking what was the problem with the java sol

File: solutions/cses-1090.mdx [hunk 5]
```diff
@@ -84,48 +85,44 @@ int main(){
...
+		Scanner sc = new Scanner(System.in);
+
+		int n = sc.nextInt();
+		int x = sc.nextInt();
+		sc.nextLine();
+
+		// Read in weights of children (this optimization is necessary to pass 
+		// all test cases)
+		String[] weightsStr = sc.nextLine().split(" ");
+		Integer[] weights = new Integer[n];
```
Comment (SansPapyrus683) [inline solutions/cses-1090.mdx:102]: is using `Integer` really necessary

File: solutions/cses-1090.mdx [hunk 6]
```diff
@@ -43,7 +44,8 @@ const int maxn=2e5+10;
 
 //variables used for the current problem
```
Comment (SansPapyrus683) [inline solutions/cses-1090.mdx:45]: can you put a space after the `//` for comments please and thank you

File: solutions/cses-1090.mdx [hunk 7]
```diff
@@ -81,4 +83,49 @@ int main(){
...
+		String[] weightsStr = sc.nextLine().split(" ");
+		Integer[] weights = new Integer[n];
+		for (int i = 0; i < n; i++) {
+			weights[i] = Integer.parseInt(weightsStr[i]);
+		}
+		Arrays.sort(weights);
+
+		int ans = 0;
+		int i = 0;  // left pointer
+		int j = n-1;  // right pointer
```
Comment (SansPapyrus683) [inline solutions/cses-1090.mdx:110]: space between ops

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
honestly p nice

File: solutions/cses-1090.mdx (hunk 5 above)
Comment (nathangong) [reply solutions/cses-1090.mdx:102]: yes, if I don't it would TLE because Integer.parseInt() returns an Integer and casting it to an int takes too long

**Comment by nathangong** [comment]
> just asking what was the problem with the java sol
It would TLE on one test case

File: solutions/cses-1090.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/cses-1090.mdx:101]: maybe add that as a comment to the code

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ar492 — APPROVED** [review]

**Pull request merged**